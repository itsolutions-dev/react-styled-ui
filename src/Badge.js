import styled, { css } from './styledComponent';
import {
  getTextColor,
  getBackgroundColor,
} from './utils/';

// TODO : get font size from theme: small, medium, large

const Badge = styled({
  component: 'div',
  style: css`
    color: ${props => getTextColor(props)};
    background: ${props => getBackgroundColor(props)};
    font-size: 12px;
    padding: 2px 6px;
    font-weight: bold;
    display: inline-block;
    vertical-align: middle;
    border-radius: ${props => props.theme.radius};
  `,
});

export default Badge;
