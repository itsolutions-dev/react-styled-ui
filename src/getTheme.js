import defaultTheme from './theme';
import deepMerge from './utils/deepMerge';

const getTheme = deepMerge.bind(null, defaultTheme);

export default getTheme;
