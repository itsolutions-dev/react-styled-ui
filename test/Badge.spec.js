import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import getTheme from '../src/getTheme';
import Badge from '../src/Badge';

const theme = getTheme({
  radius: '7px',
});

describe('Badge', () => {
  test('should render a Badge', () => {
    const component = ReactTestRenderer.create(<Badge theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<Badge theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have a border-radius', () => {
    const component = ReactTestRenderer.create(
      <Badge
        theme={theme}
      />,
    );
    expect(component).toHaveStyleRule('border-radius', '7px');
  });
});
