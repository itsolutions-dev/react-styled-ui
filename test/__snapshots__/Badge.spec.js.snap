// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render a Badge 1`] = `
.c0 {
  color: #4c4c4c;
  background: #ffffff;
  font-size: 12px;
  padding: 2px 6px;
  font-weight: bold;
  display: inline-block;
  vertical-align: middle;
  border-radius: 7px;
}

<div
  className="c0"
/>
`;
